### PR TITLE
Fix invalid date and age ranges in reports (Fixes #1499)

### LINKED ISSUE #1499: Invalid Date Range Accepted Within Report Forms
A) **Outreachy Username:** maggie.negm
B) **Issue Title:** Invalid Date Range Accepted Within Report Forms
C) **Report Date:** 8/19/2019
D) **Sites Affected:** Documentation Site
E) **OS/Browser Used:** Chrome browser in Linux OS
F) **EHR Workflow Module:** Report modules that accept a date range within their forms, including Report – Patient List, Reports – Patient List by Reporter, Report – Patient List Creation, Report – Clinical, Report – Immunization Registry, and more
G) **Steps to Reproduce the Issue:**
All the Report modules exist under the “Reports” tab in the main menu. 

This example uses the “Report – Patient List” module:
1. Select “Reports” from the main menu.
2. Select “Clients” under “Reports.”
3. Select “Patient List” under “Clients.”
4. Enter an invalid date range, such as a From date that is before the To date, into the form within the “Report – Patient List” module. (You can choose or not choose a specific provider. It won’t affect the results, since filtering within this module doesn’t work correctly as of today. There is an open issue regarding this: https://github.com/LibreHealthIO/lh-ehr/issues/1496.)

H) **Expected Behavior:** An expected scenario is to have the form prevent users from entering an invalid date range and to output a message that informs users of their mistake and how to fix it.
I) **What Actually Happened:** The form accepted the invalid date range and output a list of patients. (It is important to mention that currently the “Report – Patient List” module isn’t filtering information correctly and displays the same list of patients no matter the input.)
J) **Screenshots with Explanation:**
The form was submitted with the following input:
Provider: Virginia Apgar
From: 08/01/2019
To: 01/01/2019
This was the output:
w-patient-from-and-to-dates-selected-results
K) **Estimation of Severity:** Critical – module function continues with ongoing errors

Similar Issue: #1415

### PR DESCRIPTION
**Pull request by mhn94**
Fixes Issue #1499.

Forms within the Reports tab no longer accept invalid date and age ranges by first calling the appropriate check function within report_validation.js. If the range is invalid, the form doesn't submit and an appropriate message appears to let the user know. If the range is valid, the forms submits as normal.

Fields are also repopulated after forms are submitted.

### CODE REVIEW

**Comment by KoniKodes** [comment]
This really did fix most of the Reports section. I'm glad j-pro volunteered to document that section. Thank you for your hard work @mhn94 
@muarachmann @Ngai-E Could you check for merge please?

File: interface/orders/procedure_stats.php [hunk 1]
```diff
@@ -264,6 +264,18 @@ function process_result_code($row) {
  .detail    { color:#000000; font-family:sans-serif; font-size:10pt; font-weight:normal }
 </style>
 <script type="text/javascript" src="../../library/textformat.js"></script>
+<script type="text/javascript" src="../../library/js/jquery-1.9.1.min.js"></script>
+<script type="text/javascript" src="../../library/report_validation.js"></script>
+
+<?php
+  call_required_libraries(array("iziModalToast"));
```
Comment (muarachmann) [inline interface/orders/procedure_stats.php]: calling jquery here and please use the latest version of the assets i think `3.3.1` if I am not mistaken

File: interface/reports/adjustments.php [hunk 2]
```diff
@@ -35,6 +35,11 @@
   <script type="text/javascript" src="<?php echo $GLOBALS['webroot'] ?>/library/overlib_mini.js"></script>
   <script type="text/javascript" src="<?php echo $GLOBALS['webroot'] ?>/library/textformat.js"></script>
   <script type="text/javascript" src="<?php echo $GLOBALS['webroot'] ?>/library/js/jquery-1.9.1.min.js"></script>
```
Comment (muarachmann) [inline interface/reports/adjustments.php]: Replace with latest version in codebase

File: interface/reports/collections_report.php [hunk 3]
```diff
@@ -65,6 +65,11 @@
 </style>
 
 <script type="text/javascript" src="../../library/js/jquery-1.9.1.min.js"></script>
```
Comment (muarachmann) [inline interface/reports/collections_report.php]: same here

File: interface/orders/procedure_stats.php (hunk 1 above)
Comment (mhn94) [reply interface/orders/procedure_stats.php]: Should I fix any other instances in other files that reference the older version of jQuery?

Comment (muarachmann) [reply interface/orders/procedure_stats.php]: Hi @mhn94 yes but if you feel like it slows you down just fix the ones with the files in this PR

Comment (muarachmann) [reply interface/orders/procedure_stats.php]: Hi @mhn94 yes but if you feel like it slows you down just fix the ones with the files in this PR

**Pull request merged**